packages/typespec-vscode/test/extension/common/commonSteps.ts

```diff
@@ -0,0 +1,343 @@
+  await retry(
+    page,
+    5,
+    async () => {
+      listForCreate = page
+        .locator("a")
+        .filter({ hasText: `TypeSpec: ${command}` })
+        .first()
+      return (await listForCreate.count()) > 0
+    },
+    "Failed to find the specified option"
+  )
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/input_command.png") })
+  await listForCreate!.click()
+}
+
+/**
+ * Start the Project with Right click on the file
+ * @param page vscode object
+ * @param command create, emit or import
+ * @param type specify whether the click is on file, folder or empty folder
+ * command: specify which command to execute to the project
+ */
+async function startWithRightClick(page: Page, command: string, type?: string) {
+  if (
+    command == "Emit from TypeSpec" ||
+    command == "Preview API Documentation"
+  ) {
+    const target = page.getByRole("treeitem", { name: "main.tsp" }).locator("a")
+    await target.click({ button: "right" })
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/click_main.png") })
+    await page.getByRole("menuitem", { name: command }).click()
+    await page.screenshot({ path: path.resolve(__dirname, `../../images-linux/` +
+      `${command == "Emit from TypeSpec" ? "emit" : "preview"}_typespec.png`)
+    })
+  } else if (command == "Import TypeSpec from Openapi 3") {
+    const targetName =
+      type === "emptyfolder"
+        ? "ImportTypespecProjectEmptyFolder"
+        : "openapi.3.0.yaml"
+    const target = page.getByRole("treeitem", { name: targetName }).locator("a")
+    await target.click({ button: "right" })
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/openapi.3.0.png") })
+    await sleep(3)
+    await page
+      .getByRole("menuitem", { name: "Import TypeSpec from OpenAPI" })
+      .click()
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/import_typespec.png") })
+  }
+}
+
+/**
+ * In vscode, when you need to select a folder or a file, call this method
+ * @param file When selecting a file, just pass it in. If you need to select a folder, you do not need to pass this parameter in.
+ */
+async function selectFolder(page: Page, file: string = "") {
+  await sleep(10)
+  await keyboard.type(file)
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/select_folder.png") })
+  await keyboard.pressKey(Key.Enter)
+  await keyboard.releaseKey(Key.Enter)
+  await sleep(3)
+}
+
+/**
+ * If the current folder is not empty, sometimes a pop-up will appear
+ * asking "Do you want to continue selecting the current folder as the root directory?".
+ * In this method, select "yes" because selecting "no" does not make sense.
+ * @param page vscode object
+ */
+async function notEmptyFolderContinue(page: Page) {
+  let yesBtn: Locator
+  await retry(
+    page,
+    5,
+    async () => {
+      yesBtn = page.locator("a").filter({ hasText: "Yes" }).first()
```

Comment:
do those not have roles or aria attributes you can query by. Querying by tag is more likely to be broken as it is an implementation detail.

Reply:
It has been updated to use roles for querying.